### PR TITLE
06305 Browser and Platform support noncontiguous NodeId

### LINKED ISSUE #6305: Noncontiguous NodeId Batch - Browser and Platform
Use NodeId API from AddressBook

For each batch, these are the changes that need to be made. 
- NodeId - All methods returning a long node id now return NodeId
- NodeId - All methods consuming a long node id now consume NodeId
- NodeId - Change usage of int node ids as index of arrays/lists to  Map<NodeId, ?>
- NodeId - Complete translation of all other usage of int/long node ids to NodeId

Files:
- Browser
- Platform
- PlatformTestingTool
- PlatformIdentity
- SwirldsGUI
- HashGraphDemoMain
- AccountBalanceExport
- SwirldsPlatform
- RecoveryPlatform
- Unit Tests (Related)

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
Fixes #6305 

Cannot merge until after: https://github.com/hashgraph/hedera-services/pull/6607
* likely will have merge conflicts that need to be cleaned up.

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java [hunk 1]
```diff
@@ -39,14 +41,17 @@
...
     @CommandLine.Option(
             names = {"-l", "--local-node"},
             description = "Specify a node that should be run in this JVM. If no nodes are provided, "
-                    + "all nodes with local IP addresses are loaded in this JVM.")
-    private void setLocalNodes(final List<Integer> localNodes) {
-        this.localNodes = localNodes;
+                    + "all nodes with local IP addresses are loaded in this JVM. Multiple nodes can be "
+                    + "specified by repeating the parameter `-l=#1 -l=#2 -l=#3`.")
+    private void setLocalNodes(@NonNull final Long... localNodes) {
+        for (Long nodeId : localNodes) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java:52]: Nit: `final`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java [hunk 2]
```diff
@@ -610,37 +609,41 @@ private Collection<SwirldsPlatform> createLocalPlatforms(
...
         final AddressBook addressBook = appDefinition.getAddressBook();
 
         int ownHostIndex = 0;
 
-        for (int i = 0; i < addressBook.getSize(); i++) {
-            final Address address = addressBook.getAddress(i);
+        for (final Address address : addressBook) {
+            final NodeId nodeId = address.getNodeId();
+            final int instanceNumber = addressBook.getIndexOfNodeId(nodeId);
+            if (!appMains.containsKey(nodeId)) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:631]: Why is this check needed, if we're also checking `if (address.isOwnHost())` directly below?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java [hunk 3]
```diff
@@ -610,37 +609,41 @@ private Collection<SwirldsPlatform> createLocalPlatforms(
...
+                // this node is not set to run locally
+                continue;
+            }
 
             if (address.isOwnHost()) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:636]: Is this intentionally excluding addresses that are part of `localNodesToStart`, but `address.isOwnHost()` returns false?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:631]: You're right the calculation of the `appMains` includes a check of `address.isOwnHost()`.  I can remove the later check.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java (hunk 3 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:636]: I am removing this.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/stream/QueueThreadObjectStreamConfiguration.java [hunk 4]
```diff
@@ -202,7 +206,9 @@ public long getNodeId() {
      *
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/stream/QueueThreadObjectStreamConfiguration.java:206]: removed

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractThreadConfiguration.java [hunk 5]
```diff
@@ -485,8 +488,9 @@ public Long getNodeId() {
      * @return this object
      */
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/framework/internal/AbstractThreadConfiguration.java:489]: removed portion of comment on allowing no nodeId.  It should just not be set if null is being set.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java:52]: made final

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java [hunk 6]
```diff
@@ -47,7 +47,7 @@ public class BrowseCommand extends AbstractCommand {
             names = {"-l", "--local-node"},
             description = "Specify a node that should be run in this JVM. If no nodes are provided, "
                     + "all nodes with local IP addresses are loaded in this JVM. Multiple nodes can be "
-                    + "specified by repeating the parameter `-l=#1 -l=#2 -l=#3`.")
+                    + "specified by repeating the parameter `-l #1 -l #2 -l 3`.")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java:50]: You removed the `#` in front of the `3`

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java:50]: whoops :-)

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/BrowseCommand.java]: fixed

**Review by alittley — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**